src/util/getNextThoughtsWithContextChain.js

```diff
@@ -0,0 +1,178 @@
+import { store } from '../store.js'
+
+import {
+  RANKED_ROOT,
+  ROOT_TOKEN,
+} from '../constants.js'
+// util
+import {
+  contextOf,
+  getContextsSortedAndRanked,
+  getSortPreference,
+  getThought,
+  getThoughtsRanked,
+  getThoughtsSorted,
+  head,
+  isFunction,
+  meta,
+  pathToContext,
+  splitChain,
+  nextSibling as thoughtNextSibling,
+  unroot,
+} from '../util.js'
+import { isContextViewActive } from './isContextViewActive.js'
+
+const contextToThought = (context, thoughtIndex) => context.map(contextValue => {
+  const thought = getThought(contextValue, thoughtIndex)
+  return { value: thought.value, rank: thought.contexts[0] && thought.contexts[0].rank }
+})
+
+const mapContextsToThoughts = (contexts, thoughtIndex) => contexts.map(({ context }) => contextToThought(context, thoughtIndex))
+
+const isValidContextView = context => {
+  // const allowSingleContext = false
+  if (!context.length) {
+    return false
+  }
+  return isContextViewActive(context) && getContextsSortedAndRanked(head(pathToContext(context))).length > 1
+}
+
+const getContextSiblings = (context, thoughtIndex) => {
+  const contexts = getContextsSortedAndRanked(head(context))
+  return mapContextsToThoughts(contexts, thoughtIndex)
+}
+
+const nextSiblingContext = (value, rank, context, thoughtIndex) => {
+  const contextSiblings = getContextSiblings(context, thoughtIndex)
+  const currentIndex = contextSiblings.findIndex(thoughts => thoughts.find(thought => thought.value === value))
+  return contextSiblings[currentIndex + 1]
+}
+
+const firstChildOfContext = (value, cursor, thoughtIndex) => {
+  const context = pathToContext(cursor)
+  const contextSiblings = getContextSiblings(context, thoughtIndex)
+  return contextSiblings[0]
+}
+
+const firstChildOfThought = (thoughtsRanked, showHiddenThoughts) => {
+  const contextMeta = meta(thoughtsRanked)
+  const sortPreference = getSortPreference(contextMeta)
+  const children = (sortPreference === 'Alphabetical' ? getThoughtsSorted : getThoughtsRanked)(thoughtsRanked)
+  const notHidden = child => !isFunction(child.value) && !meta(pathToContext(thoughtsRanked).concat(child.value)).hidden
+  const childrenFiltered = showHiddenThoughts ? children : children.filter(notHidden)
+  return childrenFiltered[0]
+}
+
+const getDerivedCursor = contextChain => {
+  // last of second last item in context chain gives us the current context
+  const context = head(contextChain.slice(-2, -1)[0])
+  const contexts = getContextsSortedAndRanked(context.value)
+  const currentContextTop = head(contextChain)[0].value
+  const matchedContextChain = contexts.find(c => c.context.includes(currentContextTop))
+  const matchedContext = matchedContextChain.context.map(i => ({ value: i, rank: matchedContextChain.rank }))
+
+  return [...matchedContext, context, ...head(contextChain).slice(1)]
+}
+
+const contextViewLookup = (value, context, rank, cursor, contextRanked, contextChain, ignoreChildren) => {
```

Comment:
What is a "Lookup"? This name does not give a clear picture of what this function does. What does the function do, and what does it return? This needs to be clearly documented.

Reply:
Updated the names and documented in the description